core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -418,7 +421,11 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
                         throw new VersionConflictEngineException(shardId, get.type(), get.id(),
                             get.versionType().explainConflictForReads(versionValue.version, get.version()));
                     }
+                    if (onRefreshMetric == null)
```

Comment:
I'd name it `onRefresh` because the engine doesn't want to know about metrics at all, even in names.